File: xbo_mmh_custom_pos_receipt/static/src/app/models/res_partner.js
/** @odoo-module **/

import { patch } from "@web/core/utils/patch";
import { ResPartner } from "@point_of_sale/app/models/res_partner";


// Patch ResPartner model to include the patient_type_id field for POS
patch(ResPartner, {
    fields: {
        ...ResPartner.fields,
        patient_type_id: {
            type: "many2one",
            relation: "patient.type",
        },
    },
});

File: xbo_mmh_custom_pos_receipt/models/res_partner.py
# -*- coding: utf-8 -*-

from odoo import models

class ResPartner(models.Model):
    _inherit = 'res.partner'

    # This function is used to add the custom 'patient_type_id' field from res.partner for use in the POS
    def _load_pos_data_fields(self, config_id):
        fields = super()._load_pos_data_fields(config_id)
        fields += [
            "patient_type_id",
            "mr_number",
        ]
        return fields

